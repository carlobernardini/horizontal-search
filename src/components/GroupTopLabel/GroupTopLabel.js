import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const GroupTopLabel = React.forwardRef(({ children, shown }, ref) => (
    <span
        className={ classNames('orizzonte__group-label--top', {
            'orizzonte__group-label--top-shown': shown
        }) }
        ref={ ref }
    >
        { children }
    </span>
));

GroupTopLabel.propTypes = {
    children: PropTypes.string,
    shown: PropTypes.bool
};

GroupTopLabel.defaultProps = {
    children: null,
    shown: false
};

export default GroupTopLabel;
