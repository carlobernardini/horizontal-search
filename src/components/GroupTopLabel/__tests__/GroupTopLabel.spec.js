import React from 'react';
import GroupTopLabel from '../GroupTopLabel';

describe('<GroupTopLabel />', () => {
    it('should render a hidden group top label', () => {
        const wrapper = shallow(
            <GroupTopLabel>
                Test
            </GroupTopLabel>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a shown group top label', () => {
        const wrapper = shallow(
            <GroupTopLabel
                shown
            >
                Test
            </GroupTopLabel>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
