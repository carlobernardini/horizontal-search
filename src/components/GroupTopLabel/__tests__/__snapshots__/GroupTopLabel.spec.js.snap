// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupTopLabel /> should render a hidden group top label 1`] = `
<span
  className="orizzonte__group-label--top"
>
  Test
</span>
`;

exports[`<GroupTopLabel /> should render a shown group top label 1`] = `
<span
  className="orizzonte__group-label--top orizzonte__group-label--top-shown"
>
  Test
</span>
`;
